src/Compiler/Microsoft.CodeAnalysis.Razor/src/ComponentTagHelperDescriptorProvider.cs

```diff
@@ -452,9 +513,11 @@ private static void CreateContextParameter(TagHelperDescriptorBuilder builder, s
     // - have the [Parameter] attribute
     // - have a setter, even if private
     // - are not indexers
-    private static Dictionary<string, (IPropertySymbol property, PropertyKind kind)>.ValueCollection GetProperties(INamedTypeSymbol type)
+    private static (IPropertySymbol property, PropertyKind kind)[] GetProperties(INamedTypeSymbol type)
```

Comment:
Wouldn't it be possible to return ImmutableArray?

Reply:
Good catch! Thanks!